client/src/pages/Home.js

```diff
@@ -1,11 +1,26 @@
 import { useEffect, useState } from "react";
 import { Link } from "react-router-dom";
 import MainTable from "../Component/MainTable";
+import SMSModal from "../Component/SMSModal";
 
 import "./Home.css";
 
+
+
 export function Home() {
 	const [message, setMessage] = useState("Loading...");
+	const [checkedCheckboxes, setChecked] = useState([]);
+
+	const onHandleSelectedUserState = (isSelected, userID ) => {
+		if (isSelected === true) {
+			checkedCheckboxes.push(userID);
+			setChecked([...checkedCheckboxes]);
```

Comment:
```suggestion
			setChecked((previousCheckedCheckboxes) => {
				return previousCheckedCheckboxes.concat(clientId);
			});
```
`checkedCheckboxes` is a React state variable, and we should never directly update it like we're doing here with `push`. Instead, we need to update it in its setter `setChecked`, and we can get its existing value as the first parameter of the callback. Then we can use `concat` which returns a new array and leaves `previousCheckedCheckboxes` unmodified